### PR TITLE
Add `Rootstock` network

### PR DESCRIPTION
**Pull request by ahsan-javaiid**
### Description
This PR adds `rootstock` network to the rainbow wallet. Rootstock is bitcoin sidechain with full compatibility with ethereum.

Website: https://rootstock.io
Defillama: https://defillama.com/chain/Rootstock
Explorer: https://explorer.rootstock.io
Developer portal: https://dev.rootstock.io 

### Demo
Screenshot 2024-04-18 at 7 02 36 PM
Screenshot 2024-04-18 at 7 04 17 PM
Screenshot 2024-04-18 at 7 05 03 PM
Screenshot 2024-04-18 at 7 04 48 PM
Screenshot 2024-04-18 at 7 03 32 PM
Screenshot 2024-04-18 at 7 02 51 PM

### Testing
- Rootstock network
- Send / Receive RBTC
- See balance of RBTC and ERC20 Tokens

### CODE REVIEW

**Comment by ahsan-javaiid** [comment]
cc: @brunobar79 @derHowie @DanielSinclair @estebanmino review. Feel free to let me know if anything needs to be changes. Looking forward to adding Rootstock in rainbow.

File: src/entries/popup/pages/settings/customChain/index.tsx
Comment (DanielSinclair) [inline src/entries/popup/pages/settings/customChain/index.tsx:1]: Please add Rootstock mainnet here

**Review by DanielSinclair — CHANGES_REQUESTED** [review]
Hello! This unfortunately won't add full support for the network as it is only touching on the frontend pieces. Instead, please revert these changes, and add Rootstock mainnet alongside the testnet to add it to our Add Network flow.

File: src/entries/popup/pages/settings/customChain/index.tsx
Comment (ahsan-javaiid) [reply src/entries/popup/pages/settings/customChain/index.tsx:1]: Thank you @DanielSinclair for the reviewing this PR. I would like to ask a quick question:
I am looking to add `Rootstock` as a rainbow supported network in wallet. That's why i added all these changes to add Rootstock network in the list of available rainbow supported networks.   

I know we can add it as custom network as you suggested. 

Asking, if you can please consider this request to add Rootstock as rainbow supported network and i can do any further changes requested.

I will look forward to your thoughts.

Comment (DanielSinclair) [reply src/entries/popup/pages/settings/customChain/index.tsx:1]: @ahsan-javaiid That unfortunately won't be possible as we require significant demand and vendor support before we can roll out support for any particular chain as a "first class" network. You can however add chain information to our Add Network flow to make it easier for users to add that network.

Comment (ahsan-javaiid) [reply src/entries/popup/pages/settings/customChain/index.tsx:1]: Hi @DanielSinclair 

Please share your telegram id. I will connect you and your team with Rootstock to provide full vendor support and meet the other significant demands. We can discuss rest of the requirements there and start working on adding as "first class" network into rainbow. 

I would be thankful to you.

Comment (DanielSinclair) [reply src/entries/popup/pages/settings/customChain/index.tsx:1]: @ahsan-javaiid Prospective chains can email `partnerships@rainbow.me`, but the better option will be to revert some of the changes here and add the mainnet chain to our Add Network flow.

**Comment by DanielSinclair** [comment]
Closing in favor of #1508

File: src/entries/popup/pages/settings/customChain/index.tsx
Comment (ahsan-javaiid) [reply src/entries/popup/pages/settings/customChain/index.tsx:1]: Hi @DanielSinclair
Hope you are doing well.
Thanks for adding Rootstock as custom network and making a release last time.
Would be great to hear back on partnership email sent on `partnerships@rainbow.me` sent by Rootstock team. 
Thank you.

Comment (ahsan-javaiid) [reply src/entries/popup/pages/settings/customChain/index.tsx:1]: @@DanielSinclair Any thought's on partnership here to complete the full integration. Would be happy to connect and discuss. Hope to hear back from your side. 🙂

Comment (ahsan-javaiid) [reply src/entries/popup/pages/settings/customChain/index.tsx:1]: @DanielSinclair Any reply on partnership email. Hope to hear back from your side. 

Visit Rootstock network stats: https://defillama.com/chain/Rootstock
Website: https://rootstock.io/
Explorer: https://explorer.rootstock.io/
Developer portal: https://dev.rootstock.io/

**Pull request closed without merging**